src/ripple/core/impl/LoadEvent.cpp

```diff
@@ -18,58 +18,71 @@
 //==============================================================================
 
 #include <ripple/core/LoadEvent.h>
-#include <ripple/core/LoadMonitor.h>
 #include <cassert>
-#include <iomanip>
 
 namespace ripple {
 
-LoadEvent::LoadEvent(
-    LoadMonitor& monitor,
-    std::string const& name,
-    bool shouldStart)
-    : monitor_(monitor)
-    , running_(shouldStart)
-    , name_(name)
-    , mark_{std::chrono::steady_clock::now()}
+LoadEvent::LoadEvent(callback_t& callback, std::string name, bool shouldStart)
+    : name_(std::move(name))
+    , callback_(callback)
+    , mark_(std::chrono::steady_clock::now())
     , timeWaiting_{}
     , timeRunning_{}
+    , running_(shouldStart)
+    , neutered_(false)
 {
 }
 
-LoadEvent::~LoadEvent()
+LoadEvent::LoadEvent(LoadEvent&& other)
+    : name_(std::move(other.name_))
+    , callback_(other.callback_)
+    , mark_(other.mark_)
+    , timeWaiting_(other.timeWaiting_)
+    , timeRunning_(other.timeRunning_)
+    , running_(other.running_)
+    , neutered_(false)
 {
-    if (running_)
-        stop();
+    other.running_ = false;
+    other.neutered_ = true;
 }
 
-std::string const&
-LoadEvent::name() const
+LoadEvent&
+LoadEvent::operator=(LoadEvent&& other)
 {
-    return name_;
+    if (this != &other)
+    {
+        name_ = std::move(other.name_);
+        callback_ = other.callback_;
+        running_ = other.running_;
```

Comment:
should we also set:
`neutered_ = other.neutered_;`?

Reply:
Hmm... catching the case where we assign an already neutered object? Yeah, I think it's best if we do that.